crates/solver-order/src/implementations/standards/_7683.rs

```diff
@@ -135,6 +148,213 @@ impl Eip7683OrderImpl {
+
+		// Get the domain separator and contract address from the network config
+		let origin_chain_id = order_data.origin_chain_id.to::<u64>();
+		let network = self.networks.get(&origin_chain_id).ok_or_else(|| {
+			OrderError::ValidationFailed(format!(
+				"Chain ID {} not found in networks configuration",
+				origin_chain_id
+			))
+		})?;
+
+		// Get the compact settler address (this should be the contract address for signature validation)
+		let contract_address = {
+			let addr = network
+				.input_settler_compact_address
+				.clone()
+				.unwrap_or_else(|| network.input_settler_address.clone());
+			AlloyAddress::from_slice(&addr.0)
+		};
+
+		// Get domain separator from TheCompact contract dynamically
+		let domain_separator = self
+			.fetch_domain_separator(network, origin_chain_id)
+			.await
+			.unwrap();
+
+		// Create the signature validator
+		let validator = CompactSignatureValidator::new(domain_separator, contract_address);
+
+		// Validate the signature
+		let validation_result = validator
+			.validate_order_production(&standard_order, &signature, user_address)
+			.map_err(|e| {
+				OrderError::ValidationFailed(format!("Signature validation failed: {}", e))
+			})?;
+
+		if !validation_result.valid {
+			return Err(OrderError::ValidationFailed(
+				"Invalid signature for ResourceLock order".to_string(),
+			));
+		}
+
+		Ok(())
+	}
+
+	/// Convert Eip7683OrderData to OifStandardOrder for signature validation
+	fn convert_to_standard_order(
+		&self,
+		order_data: &Eip7683OrderData,
+	) -> Result<OifStandardOrder, OrderError> {
+		// Parse addresses
+		let user_address = AlloyAddress::from_slice(
+			&hex::decode(order_data.user.trim_start_matches("0x")).map_err(|e| {
+				OrderError::ValidationFailed(format!("Invalid user address: {}", e))
+			})?,
+		);
+
+		let input_oracle_address = AlloyAddress::from_slice(
+			&hex::decode(order_data.input_oracle.trim_start_matches("0x")).map_err(|e| {
+				OrderError::ValidationFailed(format!("Invalid input oracle address: {}", e))
+			})?,
+		);
+
+		// Convert outputs
+		let outputs: Vec<MandateOutput> = order_data
+			.outputs
+			.iter()
+			.map(|output| MandateOutput {
+				oracle: FixedBytes::from(output.oracle),
+				settler: FixedBytes::from(output.settler),
+				chainId: output.chain_id,
+				token: FixedBytes::from(output.token),
+				amount: output.amount,
+				recipient: FixedBytes::from(output.recipient),
+				call: output.call.clone().into(),
+				context: output.context.clone().into(),
+			})
+			.collect();
+
+		Ok(OifStandardOrder {
+			user: user_address,
+			nonce: order_data.nonce,
+			originChainId: order_data.origin_chain_id,
+			expires: order_data.expires,
+			fillDeadline: order_data.fill_deadline,
+			inputOracle: input_oracle_address,
+			inputs: order_data.inputs.clone(),
+			outputs,
+		})
+	}
+
+	/// Fetch domain separator from TheCompact contract dynamically
+	async fn fetch_domain_separator(
```

Comment:
Let's make this `generate_domain_separator_transaction` and pass it to the solver-service layer above?